sw/device/tock/chips/opentitan_common/src/uart.rs

```diff
@@ -115,10 +27,11 @@ pub struct UartParams {
 }
 
 impl Uart<'a> {
-    pub const fn new(base: StaticRef<UartRegisters>, clock_frequency: u32) -> Uart<'a> {
+    pub const fn new(base_addr: u32, clock_frequency: u32) -> Uart<'a> {
         Uart {
-            registers: base,
+            base_addr: base_addr,
             clock_frequency: clock_frequency,
+            dif_uart: OptionalCell::empty(),
             tx_client: OptionalCell::empty(),
             rx_client: OptionalCell::empty(),
             tx_buffer: TakeCell::empty(),
```

Comment:
At first wasn't sure whether these can  potentially result in race conditions, as they are used in the interrupt handler. However, looking into how interrupts are handled in Tock, this is not an issue.